gym/vector/sync_vector_env.py

```diff
@@ -71,8 +69,11 @@ def __init__(self, env_fns, observation_space=None, action_space=None, copy=True
         self.observations = create_empty_array(
             self.single_observation_space, n=self.num_envs, fn=np.zeros
         )
-        self._rewards = np.zeros((self.num_envs,), dtype=np.float64)
-        self._dones = np.zeros((self.num_envs,), dtype=np.bool_)
+        shape = (self.num_envs,)
+        if isinstance(self.envs[0].unwrapped, VectorEnv):
```

Comment:
Is it ever possible that this condition would be different for `self.envs[0]` and `self.envs[1]`? Also, do we actually need to unwrap it? (I'm not sure off the top of my head how wrappers interact with VectorEnvs)

Reply:
I dont see how this would only be true for some envs.. I mean, it's certainly outside the reasonable use case here.

Yeah unwrapping it is necessary, since most wrappers can work the same way for both envs and VectorEns.